### PR TITLE
bind: fix global binding hint for resource groups

### LINKED ISSUE #43736: bind: support binding hint to resource group
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```sql
create resource group rg1 RU_PER_SEC=100;
use test;
drop table if exists t;
create table t(a int, b int);
create global binding for select * from t using select /*+ resource_group(rg1) */ * from t;
select * from t;
```

### 2. What did you expect to see? (Required)
Need exec resource group `rg1` rather than exec default  resource group

### 3. What did you see instead (Required)
exec default resource group

### 4. What is your TiDB version? (Required)
7.0.0

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: close #43736, Ref #38825

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Manual test (add detailed scripts or steps below)

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by CabinfeverB — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Comment by nolouch** [comment]
ptal @glorv

**Review by glorv — APPROVED** [review]

File: planner/optimize.go [hunk 1]
```diff
@@ -279,6 +279,18 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 					sessVars.StmtCtx.AppendWarning(err)
 				}
 			}
+
+			// Override the resource group if necessary
+			// TODO: we didn't check the existence of the hinted resource group now to save the cost per query
+			if curStmtHints.HasResourceGroup {
```
Comment (glorv) [inline planner/optimize.go:285]: I think you can just move the code of L185~L192 after this `if useBinding {` to avoiding duplicate this code block twice

**Review by winoros — APPROVED** [review]

File: planner/optimize.go (hunk 1 above)
Comment (HuSharp) [reply planner/optimize.go]: Done. PTAL, thx~

**Comment by HuSharp** [comment]
@glorv delete a redundant line, plz add merge again, thx!

**Pull request merged**